workitem/workitem_repository_blackbox_test.go

```diff
@@ -510,3 +513,159 @@ func (s *workItemRepoBlackBoxTest) TestConcurrentWorkItemCreations() {
 		assert.Equal(s.T(), 0, report.failures)
 	}
 }
+
+func (s *workItemRepoBlackBoxTest) TestList() {
+	s.T().Run("list with explicit order", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.Iterations(2),
+			tf.WorkItems(10, func(fxt *tf.TestFixture, idx int) error {
+				switch idx {
+				case 0, 1, 2, 3, 4, 5, 6:
+					fxt.WorkItems[idx].Fields[workitem.SystemState] = "open"
+				default:
+					fxt.WorkItems[idx].Fields[workitem.SystemState] = "new"
+				}
+				return nil
+			}),
+		)
+		t.Run("by created descending", func(t *testing.T) {
+			// when
+			exp, _ := query.Parse(ptr.String(`{"system.state": "open"}`))
+			sort, _ := workitem.ParseSortWorkItemsBy(ptr.String("-created"))
+			res, count, err := s.repo.List(context.Background(), fxt.Spaces[0].ID, exp, nil, nil, nil, sort)
+			// then
+			require.NoError(t, err)
+			assert.Equal(t, 7, count)
+			toBeFound := id.Slice{
+				fxt.WorkItems[0].ID,
+				fxt.WorkItems[1].ID,
+				fxt.WorkItems[2].ID,
+				fxt.WorkItems[3].ID,
+				fxt.WorkItems[4].ID,
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+			}.ToMap()
+			require.Equal(t, fxt.WorkItems[0].ID, res[6].ID)
+			require.Equal(t, fxt.WorkItems[1].ID, res[5].ID)
```

Comment:
Can you please move all of these into a `for` loop?

Reply:
Fixed in 6ad821e59cd8dffbb19ec35fe49dbf66a6e0ef19